frontend/views/components/modal/Modal.vue

```diff
@@ -1,7 +1,7 @@
 <template lang='pug'>
```

Comment:
All changes in this file is irrelevant to the #2173 but a fix for below intermittent bug I encountered in `Home.vue`

![image](https://github.com/okTurtles/group-income/assets/17641213/6578e670-7b02-4450-a9c4-654d3b3c5021)

This happens because when the prompt pops up for any login-error but then `LoginModal.vue` is immediately queried. `childData` for error pop-up is lost in this case. So updated `Modal.vue` to take this into account and store the `childData` as a map of `{ 'modal-name': data }` instead.